### DISCUSSION TITLE
Possibility of porting this to another backend language

### DISCUSSION THREAD

**Discussion by ajthinking** [tinkerun/tinkerun#34] [Q&A] [answered]
Thanks for making this 🎉 
May I ask you, do you think I can modify this repo to target the ruby on rails console? If so, can you share some (rough) steps I need to take? 🙏

**Comment by billyct** [comment, marked as answer]
@ajthinking I think it's possible. for now, you can just input your rails command in the command field.

<img src="https://user-images.githubusercontent.com/1612364/132714812-1a1e0b3f-431f-4f0e-a804-68c5a64f75e0.png" alt="screenshot" width="400"/>

but I have not implemented the ruby language to the editor. it only supports the PHP language right now.

maybe you can just change the `language` to `ruby`

https://github.com/tinkerun/tinkerun/blob/master/packages/tinkerun-renderer/src/components/editor/Editor.jsx#L134

**Comment by ajthinking** [comment]
Thanks @billyct I tried pointing to `rails console` and it kind of works right of the bat, awesome 🤩 just need some fixing on the linebreaks

![image](https://user-images.githubusercontent.com/3457668/132868922-4b95ab66-ea13-4c11-961d-41b434de998c.png)
